src/reporter.test.ts

```diff
@@ -589,6 +588,100 @@ describe('reporter', () => {
     expect(reporter.forgeAndScrubRequestBody([], 100)).toEqual(undefined);
   });
 
+  test('scrubSpans missing http fields', () => {
+    const spans = [
+      {
+        info: {
+          httpInfo: {
+            host: 'host',
+            request: {}, // missing body and headers
+            response: {}, // missing body and headers
+          },
+        },
+      },
+      {
+        info: {
+          // missing request and response and host
+          httpInfo: {},
+        },
+      },
+      { info: {} }, // missing httpInfo
+      {}, // missing info
+      {
+        info: {
+          httpInfo: {
+            host: 'host',
+            // missing headers
+            request: {
+              body: '',
+            },
+            // missing body
+            response: {
+              headers: {},
+            },
+          },
+        },
+      },
+      {
+        info: {
+          httpInfo: {
+            // missing host
+            request: {
+              body: '',
+              headers: {},
+            },
+            response: {
+              body: '',
+              headers: {},
+            },
+          },
+        },
+      },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
+  });
+
+  test('scrubSpans missing http fields (stepFunction)', () => {
+    const spans = [
+      {
+        info: {
+          traceId: {
+            Root: 'Root',
+            Parent: 'Parent',
+            Sampled: '0',
+            transactionId: 'transactionId',
+          },
+          httpInfo: { host: 'StepFunction' },
+          resourceName: 'StepFunction',
+          messageId: 'messageId',
+        },
+      },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
+  });
+
+  test('scrubSpans should not fail after throwing an error', () => {
+    const shouldScrubDomainMock = jest
+      .spyOn(utils, 'shouldScrubDomain')
+      .mockReturnValueOnce(true)
+      .mockReturnValueOnce(true)
+      .mockImplementationOnce(() => {
+        throw new Error('Error');
+      });
+    const beforeScrub = [
+      { info: { httpInfo: { request: { host: 'StepFunction' }, response: {} } } },
+      { info: { httpInfo: { request: {}, response: { body: 'body' } } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+    ];
+    const spans = [...beforeScrub];
+    scrubSpans(spans);
+
+    expect(spans).toEqual(beforeScrub);
```

Comment:
I don't get it - if you fail to scrub, then you just send it like that?
I think that we better remove it